group-coordinator/src/main/java/org/apache/kafka/coordinator/group/streams/StreamsGroup.java

```diff
@@ -231,6 +237,9 @@ public StreamsGroup(
         this.currentWarmupTaskToProcessIds = new TimelineHashMap<>(snapshotRegistry, 0);
         this.topology = new TimelineObject<>(snapshotRegistry, Optional.empty());
         this.configuredTopology = new TimelineObject<>(snapshotRegistry, Optional.empty());
+        this.assignmentConfigs = new TimelineHashMap<>(snapshotRegistry, 0);
+        // Set default assignment configuration
```

Comment:
I would put any default assignment configs here. We will update the value inside `GroupMetadataManager`.